### PR TITLE
Rework CSV uploader permissions

### PR DESCRIPTION
**Pull request by mehansen**
# BACKEND PULL REQUEST

## Related Issue

Customer reported issue accessing upload history page and shared the following screenshot:
image
In Okta they have the following groups:
* NO_ACCESS
* ADMIN

Admin and standard users should be able to use the uploader but currently don't have correct permissions for GraphQL queries.
They can successfully upload because that doesn't use a mutation and we check permissions here, which was updated.
https://github.com/CDCgov/prime-simplereport/blob/118d331cc8415890e457e01b6e81b2e781bce1a2/backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java#L71-L72

Also on the frontend we were not showing all standard users the message when the queue is empty recommending the bulk upload tool.
See [emptyQueueMessage function](https://github.com/CDCgov/prime-simplereport/blob/0dc1ee92cb9f476310aa2f6c4cc7ab0a3b1aa2b6/frontend/src/app/testQueue/TestQueue.tsx#L37).

## Changes Proposed

I know we originally gave everyone with read result list permissions the ability to use the uploader, but I think decoupling the permissions makes more sense in case we ever want to change them since the functionality itself is completely separated and it seems like it's going to remain that way (not planning on adding bulk upload results to results page view).
So backend changes:
* rename the SR_CSV_UPLOADER_PILOT permission to be UPLOAD_RESULTS_SPREADSHEET and give it to standard users and admin users
* leave the uploader specific permission check where we had it on the queries and add it to the Test Result Upload Service for submitting a CSV. basically on the backend you need uploader specific permissions to do upload things, not just result viewing permissions
* I'm hesitant to remove the TEST_RESULT_UPLOAD_USER role because I'm not sure how Okta will react when loading that role for users who we already assigned it to
* the tests in TestResultUploadServiceTest.java still pass without having to change anything

Frontend changes:
* keep using canViewResults for permissions for now, I think we should decouple it going forward though
* fix the message for the empty queue so anyone who can access the uploader pages from the frontend can see it (see [this PR](https://github.com/CDCgov/prime-simplereport/pull/4396) for when we made uploader accessible to anyone with view test results permissions)
* this is technically not backwards compatible because an old version of the frontend could look for the UserPermission I deleted on the backend, but it will only affect people originally in the pilot program and it should only mean they don't see the upload link in the empty queue message until they get a new version of the frontend. they know where the uploader is so I don't think it's a concern

tldr: keep the uploader permissions decoupled from viewing results on the backend, but give it to admins and standard users

## Additional Information

probably should clean up the `TEST_RESULT_UPLOAD_USER` enum in `OrganizationRole.java` but not sure how Okta will react

## Testing

Deploying to `dev`. Test as an ADMIN or STANDARD_USER you can view Upload history or view a specific upload submission. Verify existing auto coverage is sufficient for other use cases as well (namely, that TEST_ONLY users still can't use the uploader)

## Checklist for Primary Reviewer
- [x] Any large-scale changes have been deployed to `test`, `dev`, or `pentest` and smoke tested
- [x] GraphQL schema changes are backward compatible with older version of the front-end
- [ ] Changes comply with the SimpleReport Style Guide

### CODE REVIEW

**Comment by mehansen** [comment]
smoke tested in `dev` by setting up my Okta groups to be an admin (not super admin), deploying main and reproducing the error, and then deploying my branch and successfully viewing the upload submissions history

**Comment by emmastephenson** [comment]
> probably should clean up the `TEST_RESULT_UPLOAD_USER` enum in `OrganizationRole.java` but not sure how Okta will react

@mehansen I was curious about this, so did some testing in `dev` with a fake role group to see what the app does. Turns out everything works just fine, we do get a warning in the logs to the effect that the role isn't recognized, but the app itself doesn't crash or even note anything unusual in the frontend. (This is where the exception gets thrown: https://github.com/CDCgov/prime-simplereport/blob/92392edc91d0f01d487961976ad0b031c34437bb/backend/src/main/java/gov/cdc/usds/simplereport/config/authorization/OrganizationExtractor.java#L82-L82) 

Screen Shot 2022-10-31 at 9 10 33 AM


It can wait for a followup PR, but I would definitely encourage us to remove the `TEST_RESULT_UPLOAD_USER` enum as it contributes to our group creation API thresholds. We've nearly hit the rate limits for that before during Omicron when many new organizations were being created at once, so removing it will just mean there's one less group created per organization signup.

**Review by emmastephenson — APPROVED** [review]
Changes LGTM, I agree that having a dedicated user permission for the results uploader is the best approach.

File: backend/src/main/resources/graphql/main.graphqls [hunk 1]
```diff
@@ -448,14 +448,14 @@ type Query {
...
-    @requiredPermissions(allOf: ["SR_CSV_UPLOADER_PILOT"])
+    @requiredPermissions(allOf: ["UPLOAD_RESULTS_SPREADSHEET"])
   uploadSubmissions(
     startDate: DateTime
     endDate: DateTime
     pageNumber: Int
     pageSize: Int
   ): UploadSubmissionPage!
-    @requiredPermissions(allOf: ["SR_CSV_UPLOADER_PILOT"])
+    @requiredPermissions(allOf: ["UPLOAD_RESULTS_SPREADSHEET"])
```
Comment (zdeveloper) [inline backend/src/main/resources/graphql/main.graphqls:458]: Sorry about forgetting to remove this!

**Review by zdeveloper — APPROVED** [review]
Thanks for the fix and the cleanup!

**Comment by mehansen** [comment]
> > probably should clean up the `TEST_RESULT_UPLOAD_USER` enum in `OrganizationRole.java` but not sure how Okta will react
> 
> @mehansen I was curious about this, so did some testing in `dev` with a fake role group to see what the app does. Turns out everything works just fine, we do get a warning in the logs to the effect that the role isn't recognized, but the app itself doesn't crash or even note anything unusual in the frontend. (This is where the exception gets thrown:
> 
> https://github.com/CDCgov/prime-simplereport/blob/92392edc91d0f01d487961976ad0b031c34437bb/backend/src/main/java/gov/cdc/usds/simplereport/config/authorization/OrganizationExtractor.java#L82-L82
> 
> )
> Screen Shot 2022-10-31 at 9 10 33 AM
> 
> It can wait for a followup PR, but I would definitely encourage us to remove the `TEST_RESULT_UPLOAD_USER` enum as it contributes to our group creation API thresholds. We've nearly hit the rate limits for that before during Omicron when many new organizations were being created at once, so removing it will just mean there's one less group created per organization signup.

thanks for testing this! I was a little swamped with on-call stuff and wanted to get this out quickly for the support request. I'll open a PR to remove it right now
